torch/csrc/distributed/rpc/init.cpp

```diff
@@ -500,6 +500,23 @@ PyObject* rpc_init(PyObject* /* unused */) {
               The number of threads in the thread-pool used by
               :class:`~torch.distributed.rpc.TensorPipeAgent` to execute
               requests.
+          )")
+      .def_readwrite(
+          "map_locations",
+          &TensorPipeRpcBackendOptions::mapLocations,
+          R"(The device map locations.)")
+      .def(
+          "set_map_location",
+          &TensorPipeRpcBackendOptions::setMapLocation,
+          R"(
+              Set device mapping between each RPC caller and callee pairs. This
+              function can be called multiple times to incrementally update
+              device placement configurations.
+
+              Arguments:
```

Comment:
What happens if I call this function when `worker_name` == this worker? Should it work as expected, i.e. if I do an RPC to self and want to map tensors from cuda:0 to cuda:1, will the RPC function run with the tensors on cuda:1? Should we add a test for this and document what's expected/

Reply:
This should work, but let me add a test for it.